resteasy-core/src/main/java/org/jboss/resteasy/specimpl/AbstractBuiltResponse.java

```diff
@@ -212,19 +212,21 @@ public void setMetadata(MultivaluedMap<String, Object> metadata)
 
    public void addMethodAnnotations(Annotation[] methodAnnotations)
    {
-      List<Annotation> ann = new ArrayList<Annotation>();
       if (annotations != null)
       {
+         List<Annotation> ann = new ArrayList<Annotation>();
          for (Annotation annotation : annotations)
          {
             ann.add(annotation);
          }
+         for (Annotation annotation : methodAnnotations)
+         {
+            ann.add(annotation);
+         }
+         annotations = ann.toArray(new Annotation[ann.size()]);
+      } else {
+         annotations = methodAnnotations;

```

Comment:
This could be simplified to:
```
public void addMethodAnnotations(Annotation[] methodAnnotations)
{
   if (annotations != null)
   {
      if (methodAnnotations != null) {
         final int currentLen = annotations.length;
         annotations = Arrays.copyOf(annotations, currentLen + methodAnnotations.length);
         System.arraycopy(methodAnnotations, 0, annotations, currentLen, methodAnnotations.length);
      }
   } else if (methodAnnotations != null) {
      annotations = Arrays.copyOf(methodAnnotations, methodAnnotations.length);
   }
  ```

Reply:
>       if (methodAnnotations != null) {
         final int currentLen = annotations.length;
         annotations = Arrays.copyOf(annotations, currentLen + methodAnnotations.length);
         System.arraycopy(methodAnnotations, 0, annotations, currentLen, methodAnnotations.length);
      }

If I understand correctly, the  Arrays.copyof() will create another array and copy , and System.arrayCopy will do another array copy when annotations and methodAnnotations are not null.   Will it be slower than only copy once with `annotations = ann.toArray(new Annotation[ann.size()])` ?

Reply:
`System.arraycopy()` should be fast as it's a native call. It's not really critical we do it this way, but it should allocate less resources as we directly use a new array instead of using a new array in the collection, then creating another new array in the `toArray()`.